app/src/main/java/io/apicurio/registry/content/extract/ContentExtractor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.content.extract;
+
+import io.apicurio.registry.content.ContentHandle;
+import io.apicurio.registry.rest.beans.EditableMetaData;
+
+/**
+ * @author Ales Justin
+ */
+public interface ContentExtractor {
```

Comment:
Why `ContentExtractor` and not `EditableMetaDataExtractor`? Maybe make it generic?

Reply:
Perhaps we'll later add new methods/stuff to it -- some more advanced extraction, etc

Reply:
That's actually fair - maybe just `MetaDataExtractor` since it's extracting meta-data.  It's definitely extracting information **from** the content though... so maybe `ContentExtractor` is ok?  Hm...